extensions/cornerstone/src/components/NavigationComponent/NavigationComponent.tsx

```diff
@@ -0,0 +1,151 @@
+import React, { useCallback, useState } from 'react';
+import { ViewportActionArrows } from '@ohif/ui-next';
+import { useSystem } from '@ohif/core/src';
+import { utils } from '../..';
+import { useViewportSegmentations } from '../../hooks';
+import { useMeasurementTracking } from '../../hooks/useMeasurementTracking';
+import { useViewportDisplaySets } from '../../hooks/useViewportDisplaySets';
+
+/**
+ * NavigationComponent provides navigation controls for viewports containing
+ * special displaySets (SR, SEG, RTSTRUCT) to navigate between segments or measurements
+ */
+function NavigationComponent({ viewportId }: { viewportId: string }) {
+  const { servicesManager } = useSystem();
+  const { segmentationService, cornerstoneViewportService, measurementService } =
+    servicesManager.services;
+
+  // Get tracking information
+  const { isTracked, trackedMeasurementUIDs } = useMeasurementTracking({ viewportId });
+  const { viewportDisplaySets } = useViewportDisplaySets(viewportId);
+  const [measurementSelected, setMeasurementSelected] = useState(0);
+  const isSRDisplaySet = viewportDisplaySets.some(displaySet => displaySet?.Modality === 'SR');
+  const cornerstoneViewport = cornerstoneViewportService.getCornerstoneViewport(viewportId);
+
+  // Get segmentation information
+  const { segmentationsWithRepresentations } = useViewportSegmentations({
+    viewportId,
+  });
+
+  const hasSegmentations = segmentationsWithRepresentations.length > 0;
+
+  // prefer segment navigation if available
+  const navigationMode = hasSegmentations
```

Comment:
Can the display set specify the navigation mode/commands so that one can add a new one entirely just be defining a new display set and assigning some extra attributes?

Reply:
It can but i propose we do it after, given the timeline